src/mix-to-mono.c

```diff
@@ -28,41 +28,45 @@ static void mix_to_mono (SNDFILE * infile, SNDFILE * outfile) ;
 int
 main (int argc, char ** argv)
 {
+	char *iname, *oname;
 	SNDFILE *infile, *outfile ;
 	SF_INFO sfinfo = { } ;
 
-	if (argc != 3)
+	if (argc != 3) {
 		usage_exit () ;
-
-	if (strcmp (argv [argc - 2], argv [argc - 1]) == 0)
-	{	printf ("Error : input and output file names are the same.\n") ;
+	} else {
+		iname = argv[1];
+		oname = argv[2];
+	}
+
+	if (strcmp(iname, "-") == 0 || strcmp(oname, "-") == 0) {
```

Comment:
The second strcmp() is not required, because the error is only raised if the names are the same.  So if the input isn't a dash and the names are the same, then we know the output also isn't a dash.  Or to put it another way, we only need to check for name collisions when the input isn't a dash, and if that is the case, an output dash won't collide with it.